CMakeLists.txt

```diff
@@ -1,56 +1,34 @@
 cmake_minimum_required(VERSION 3.19)
-project(xania)
-
-####
-# Shell out to the Makefile to get all the paths it installed for us
-set(CMAKE_MAKE_CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/config-from-make.cmake)
-execute_process(
-        COMMAND make cmake-print-deps CMAKE_CONFIG_FILE=${CMAKE_MAKE_CONFIG_FILE}
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-        RESULT_VARIABLE MAKE_CONFIG_FAIL)
-if (MAKE_CONFIG_FAIL)
-    message(FATAL_ERROR "Unable to get makefile configuration")
-endif ()
-include(${CMAKE_MAKE_CONFIG_FILE})
-####
 
 set(BUILD_VERSION "(local build)" CACHE STRING "Build version (build number only)")
 set(BUILD_FULL_VERSION "(local build)" CACHE STRING "Full string of build version (with git info)")
 set(BUILD_COMMIT_URL "https://this.is.a.local.build/1234" CACHE STRING "Full string of git URL to this build")
 
+project(xania)
+
 set(CMAKE_CXX_STANDARD 20)
-add_compile_options(-Werror -Wall -Wextra)
+add_compile_options(-Werror -Wall -Wextra -Wno-error=deprecated-declarations)
 # Regardless of the CMake settings, I always want us to have debug information available.
 add_compile_options(-g)
 
 # -static-libasan is required to prevent doorman+xania from crashing on startup (with some error about LD_PRELOAD)
 # when you build with 'make'.
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=leak -fsanitize=address -fsanitize=undefined -static-libasan")
-
-include(scripts/conan.cmake)
-
-# See Makefile - CONAN_VERSION 
-message("If the following conan check fails; please remove your .tools directory")
-conan_check(VERSION 1.43.1 REQUIRED)
-
-conan_add_remote(NAME conancenter
-        INDEX 0
-        URL https://center.conan.io)
-
-conan_cmake_autodetect(CONAN_SETTINGS)
-
-conan_cmake_install(PATH_OR_REFERENCE ${CMAKE_SOURCE_DIR}
-        SETTINGS ${CONAN_SETTINGS}
-        BUILD missing)
-
-include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
-conan_basic_setup(TARGETS)
-
-set(CMAKE_INSTALL_MESSAGE NEVER) # prevent logging installation messages
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=leak -fsanitize=address -fsanitize=undefined -static-libasan ")
 
 enable_testing()
 
+find_package(spdlog REQUIRED)
```

Comment:
How can we tell it to use a specific version of these libs?  That's something we could do with today's build.

Reply:
Oh, I see, conanfile.txt is still there. I guess it means that for anyone not relying on conan to install those 3rd party packages, they might end up using an older/wrong version of one of the packages?

Reply:
actually, you don't do it in cmake but in conan. That said, this is normally feasible in cmake, and would probably be a good idea. To be continued :)